src/Modules/Workspaces/Commands.Workspaces/GetPowerBIWorkspace.cs

```diff
@@ -131,23 +140,35 @@ public override void ExecuteCmdlet()
                 this.Filter = $"tolower(name) eq '{this.Name.ToLower()}'";
             }
 
-            if (this.All.IsPresent && this.Scope == PowerBIUserScope.Organization)
-            {
-                this.ExecuteCmdletWithAll();
-                return;
-            }
-
             if (!string.IsNullOrEmpty(this.User) && this.Scope == PowerBIUserScope.Organization)
             {
                 var userFilter = $"users/any(u: tolower(u/emailAddress) eq '{this.User.ToLower()}')";
                 this.Filter = string.IsNullOrEmpty(this.Filter) ? userFilter : $"({this.Filter}) and ({userFilter})";
             }
 
+            this.GetWorkspaces();
+        }
+
+        private void GetWorkspaces()
+        {
             using (var client = this.CreateClient())
             {
+                bool defaultingFirst = false;
+                if (this.First == default)
+                {
+                    this.First = 100;
+                    defaultingFirst = true;
+                }
+
                 var workspaces = this.Scope == PowerBIUserScope.Individual ?
                     client.Workspaces.GetWorkspaces(filter: this.Filter, top: this.First, skip: this.Skip) :
                     client.Workspaces.GetWorkspacesAsAdmin(expand: "users", filter: this.Filter, top: this.First, skip: this.Skip);
+
+                if (defaultingFirst && workspaces.Count() == 100)
+                {
+                    this.Logger.WriteWarning("Defaulting to return only the top 100 workspaces. Use -First & -Skip (or) -All to view more.");
```

Comment:
Defaulted to show top 100 workspaces. Use -First and -Skip or -All to retrieve more results.